testing/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/executer/GradleExecuter.java

```diff
@@ -352,58 +352,32 @@ public void connect(PipedInputStream snk) throws IOException {
-     * @deprecated Use {@link #expectDeprecationWarning(String)} instead.
+     * @implNote URLs to documentation should use /current/ as the version. This fixture will automatically replace it with the actual version tested.
      */
-    @Deprecated
-    GradleExecuter expectDeprecationWarning();
+    GradleExecuter expectDocumentedDeprecationWarning(String warning);
 
     /**
-     * Expects exactly the given deprecation warning.
+     * Do not call this method directly.
      *
-     * This may show up with a strikethrough in IntelliJ as if it were deprecated.  This method is still okay to use.  You can
-     * also switch to the more specific {@link #expectDocumentedDeprecationWarning(String)} if the warning includes a documentation
-     * link and you don't want to (ironically) see code testing deprecation appearing as if it itself were deprecated.
+     * @see #expectDocumentedDeprecationWarning(String)
      */
-    default GradleExecuter expectDeprecationWarning(String warning) {
-        return expectDeprecationWarning(ExpectedDeprecationWarning.withMessage(warning));
-    }
-
-    default GradleExecuter expectDeprecationWarningWithPattern(String pattern) {
-        return expectDeprecationWarning(ExpectedDeprecationWarning.withSingleLinePattern(pattern));
-    }
-
-    default GradleExecuter expectDeprecationWarningWithMultilinePattern(String pattern) {
-        return expectDeprecationWarningWithMultilinePattern(pattern, pattern.split("\n").length);
-    }
-
-    default GradleExecuter expectDeprecationWarningWithMultilinePattern(String pattern, int numLines) {
-        return expectDeprecationWarning(ExpectedDeprecationWarning.withMultiLinePattern(pattern, numLines));
-    }
-
     GradleExecuter expectDeprecationWarning(ExpectedDeprecationWarning warning);
 
     /**
-     * Expects the given deprecation warning, allowing to pass documentation url with /current/ version and asserting against the actual current version instead.
+     * Disable deprecation warning checks.
      */
-    GradleExecuter expectDocumentedDeprecationWarning(String warning);
+    GradleExecuter noDeprecationChecks();
 
     /**
-     * Expects exactly the given number of deprecation warnings. If fewer or more warnings are produced during
-     * the execution, the assertion fails.
+     * Expects a message that contains the word "deprecated" in the output.
      *
-     * @deprecated Use {@link #expectDeprecationWarning(String)} instead.
-     */
-    @Deprecated
-    GradleExecuter expectDeprecationWarnings(int count);
-
-    /**
-     * Disable deprecation warning checks.
+     * This is not intended to test for Gradle deprecation warnings. Use {@link #expectDocumentedDeprecationWarning(String)} instead.
+     *
+     * This method is used to document builds that emit deprecation messages from external tools like javac or the Kotlin compiler.
      */
-    GradleExecuter noDeprecationChecks();
+    GradleExecuter expectDeprecatedMessage(String warning);
```

Comment:
Maybe name it `expectExternalDeprecationMessage()`?

Reply:
I like that name better 👍